### PR TITLE
Rro span details renders

### PR DESCRIPTION
**Pull request by Richard-Roberts**
span details render on page on node double click

### CODE REVIEW

File: client/src/components/home.tsx [hunk 1]
```diff
@@ -14,6 +16,10 @@ import { selectSourceMap, ViewType } from './sourceMapSlice'
 const home  = () => {
 
   const sourceMap = useAppSelector(selectSourceMap)
+
+  const spanList = useAppSelector(selectSpanMap)
+
+
```
Comment (drewdunne) [inline client/src/components/home.tsx]: Few empty lines there

File: client/src/components/home.tsx [hunk 2]
```diff
@@ -14,6 +16,10 @@ import { selectSourceMap, ViewType } from './sourceMapSlice'
 const home  = () => {
 
   const sourceMap = useAppSelector(selectSourceMap)
+
+  const spanList = useAppSelector(selectSpanMap)
```
Comment (drewdunne) [inline client/src/components/home.tsx:20]: Should these variables be the same name? AKA SpanList != SpanMap?

File: client/src/components/sourceMapSlice.ts [hunk 3]
```diff
@@ -6,10 +6,12 @@ export enum ViewType {cluster, trace}
 
 const initialState: View = {
   type: ViewType.cluster,
+  data: 'placeholder'
```
Comment (drewdunne) [inline client/src/components/sourceMapSlice.ts]: can this just initialize as undefined?

File: client/src/components/span-table/spanData.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+      const time: any = new Date(startTime/1000).toString()
+      
+    //     console.log('startTime: ', time)
+    //     const entryKey = `span-data-time-entry`
+    //     // console.log(e)
+        
+      
+    
+    //     result.push([
```
Comment (lejon1220) [inline client/src/components/span-table/spanData.tsx]: are these comments required in the code still?

File: client/src/components/sourceMapSlice.ts (hunk 3 above)
Comment (drewdunne) [reply client/src/components/sourceMapSlice.ts]: It looks like, on line 14, that it's an optional parameter so probably no need for 'placeholder?

File: client/src/components/span-table/spanData.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import { useSelector } from 'react-redux';
+import { useAppDispatch } from '../../lib/hooks';
+import { selectSpanDataList } from './spanDataSlice';
+import '../../styles/spanTable.scss'
+
+const spanData= (props: any) => {
+ 
+  console.log(props) 
+  console.log('props.spanData.duration', props.spanData.duration)
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Leftover console logs? Or are these meant for future developers?

File: client/src/components/span-table/spanData.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+// const { data } = useSelector(selectSpanDataList)
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Can this be removed?

File: client/src/components/span-table/spanData.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+const jsxElements = (() => {
+  const result: Array<Array<JSX.Element>> = []
+
+  const tags = props.spanData.tags
+
+  for (let i = 0; i < tags.length; i++) {
+    const e = tags[i];
+
+    const entryKey = `span-data-tag-entry-${i}`
+    // console.log(e)
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Can this be removed?

File: client/src/components/span-table/spanData.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    result.push([
+      <div key={entryKey} className='span-data-tag-entry'>
+        <div className="span-data-entry">
+          {i}: [<span className='boldSpanTag'>Key: </span>{key}, <span className='boldSpanTag'>Type: </span>{type}, <span className='boldSpanTag'>Value: </span>{value}]
+        </div>
+      </div>
+    ])
+  }
+
+  // onClick={()=>loadSpanTableData(e.data.name)}
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Needed?

File: client/src/components/span-table/spanData.tsx [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+  
+    const warnings = props.spanData.warnings
+    
+    if (warnings === null) {return 'null'}
+
+    for (let i = 0; i < warnings.length; i++) {
+      const e = warnings[i];
+  
+      const entryKey = `span-data-tag-entry-${i}`
+      // console.log(e)
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Needed?

File: client/src/components/span-table/spanData.tsx [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+  
+    
+
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx:49]: Several empty lines

File: client/src/components/span-table/spanData.tsx [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+          <div className="span-data-entry">
+            {e}
+          </div>
+        </div>
+      ])
+    }
+    return result})()
+
+    // const jsxStartTimes = (() => {
+    //   const result: Array<Array<JSX.Element>> = []
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Leftover debugging comments?

File: client/src/components/span-table/spanData.tsx [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    
+    //     result.push([
+    //       <div key={entryKey} className='span-data-startTime-entry'>
+    //         <div className="span-data-start-entry">
+    //           {time} ms
+    //         </div>
+    //       </div>
+    //     ])
+        
+    //   })()
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: 77-91 Leftover debugging comments?

File: client/src/components/span-table/spanData.tsx [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+      <div className="span-data-entry"> <span className='boldItalicsSpan'>Warnings: </span>{jsxWarnings}</div>
+      <div className="span-data-entry"> <span className='boldItalicsSpan'>Tags: </span>{jsxElements}</div>
+    </div>
+  )
+}
+
+  
+
+
+
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: Lots of whitespace here

File: client/src/components/span-table/spanHeader.tsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+import { useAppSelector } from '../../lib/hooks';
+import { selectSearchTraceResult } from '../searchBarSlice';
+
+/** 
+   * Defines the contents of the Trace Table Headers
+   * @Remarks Manages spacing between elements via a CSS Grid defined in traceTable.scss
+   */
+
+
+
```
Comment (drewdunne) [inline client/src/components/span-table/spanHeader.tsx]: Lots of whitespace here. Comment also seems inaccurate.

File: client/src/components/span-table/spanList.tsx [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+import '../../styles/spanTable.scss'
+import SpanData from './spanData'
+import SpanResultsMap from './spanResultsMap';
+import { ViewType } from '../sourceMapSlice';
+
+
+/** 
+   * Defines the contents of the Trace Table Headers
+   * @Remarks Manages spacing between elements via a CSS Grid defined in traceTable.scss
+   */
```
Comment (drewdunne) [inline client/src/components/span-table/spanList.tsx]: Inaccurate comments

File: client/src/components/span-table/spanList.tsx [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+const spanList = () => {
+
+  const { data } = useSelector(selectSpanTableList)
+  const dispatch = useAppDispatch();
+
+  function loadNewSpanResults(type: spanViewType, spanId: String) {
+    console.log('in loadNewSpanResults')
+    dispatch(changeSpanDataView({type}))
+    // dispatch(getSpanDataAsync(spanId))
```
Comment (drewdunne) [inline client/src/components/span-table/spanList.tsx]: Leftover comments?

File: client/src/components/span-table/spanList.tsx [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  // interface Props {
+  //   traceId? : String
```
Comment (drewdunne) [inline client/src/components/span-table/spanList.tsx]: Leftover?

File: client/src/components/span-table/spanList.tsx [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+      const entryKey = `span-table-entry-${i}`
+      console.log(e)
+      console.log('duration', data[i].spanData.duration)
+
+      const spanData = 'hello'
+      const spanID = e.spanIds
+
+      result.push([
+        <div key={entryKey} className='span-pod-entry'>
+          {/* <br></br> */}
```
Comment (drewdunne) [inline client/src/components/span-table/spanList.tsx]: Leftover?

File: client/src/components/span-table/spanListSlice.tsx [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+  status: string
+  data: SpanTableData
+}
+
+//changed to any for passing props
+type SpanNames = any
+
+type SpanTableData = SpanNames[]
+
+
```
Comment (drewdunne) [inline client/src/components/span-table/spanListSlice.tsx]: Lots of whitespace here

File: client/src/components/span-table/spanListSlice.tsx [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+    console.log(url)
+
+    const response = await fetch(url)
+    const data = await response.json();
+    console.log(data);
+    return data;
+  }
+)
+/**
+   * Handles reducer logic related to Source Map View Type Updates
```
Comment (drewdunne) [inline client/src/components/span-table/spanListSlice.tsx]: Comment outdates

File: client/src/components/span-table/spanMap.tsx [hunk 21]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../lib/hooks';
+import SpanTableContent from './spanTableContent'
+import { selectSpanMap, RenderType } from './spanMapSlice'
+import '../../styles/home.scss'
+
+/**
+ * Renders a specific Source Map "View" (utilizing Cytoscape)
+ * @Views : Currently Supports both Cluster View and Trace View
```
Comment (drewdunne) [inline client/src/components/span-table/spanMap.tsx]: Leftover comment.

File: client/src/components/span-table/spanMapSlice.tsx [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+export interface Render {
+  type: RenderType,
+  data?: String,
+  id?: String
+}
+
+/**
+   * Handles reducer logic related to Source Map View Type Updates
```
Comment (drewdunne) [inline client/src/components/span-table/spanMapSlice.tsx]: Leftover Comment. Update?

File: client/src/components/span-table/spanResultsMapSlice.tsx [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+  
+}
+
+export interface Render {
+  type: spanViewType,
+  
+}
+
+/**
+   * Handles reducer logic related to Source Map View Type Updates
```
Comment (drewdunne) [inline client/src/components/span-table/spanResultsMapSlice.tsx]: Leftover comments

File: client/src/components/traceView.tsx [hunk 24]
```diff
@@ -50,29 +73,48 @@ cytoscape.use(coseBilkent);
...
             }, 10)
-            setTimeout( function(){
-              cy.$('').unselect();
-              cy.fit(cy.$(''),50);
-            }, 5000 );
+            const nodeData = node.data()
+            // setTimeout( function(){
+            //   cy.$('').unselect();
+            //   cy.fit(cy.$(''),50);
+            // }, 5000 );
```
Comment (drewdunne) [inline client/src/components/traceView.tsx]: Can these comments be cutout?

File: client/src/styles/spanTable.scss [hunk 25]
```diff
@@ -0,0 +1,156 @@
...
+      cursor: pointer
+    }
+
+.header {
+  display: flex;
+  flex: inline-block;
+  justify-content: space-between;
+  padding-top: 1.1vh;
+  // padding-right: 1vw;
+  // padding-left: 1vw;
```
Comment (drewdunne) [inline client/src/styles/spanTable.scss]: leftover comments?

File: server/controllers/TraceController.ts [hunk 26]
```diff
@@ -84,21 +84,21 @@ export class TraceController {
...
+      // }
+      // catch(err) {
+      //   return next({
+      //     log: `Error in TraceController.getCluster: ${err}`,
+      //     status: 500,
+      //     message: {
+      //       err: 'Error occured while retrieving Span Details Data in TraceController',
+      //     },
+      //   });
+      // }
```
Comment (drewdunne) [inline server/controllers/TraceController.ts]: Lots of comments here. Needed?

**Review by lejon1220 — COMMENTED** [review]
some left over comments.

File: client/src/components/span-table/spanList.tsx [hunk 27]
```diff
@@ -0,0 +1,76 @@
...
+   
+
+    result.push([<div key="span-table-spacer-entry" className='span-entry-spacer'> </div>])
+
+    for (let i = 0; i < data.length; i++) {
+      const e = data[i];
+
+      const entryKey = `span-table-entry-${i}`
+      console.log(e)
+      console.log('duration', data[i].spanData.duration)
```
Comment (drewdunne) [inline client/src/components/span-table/spanList.tsx:34]: Are These logs still needed?

File: client/src/components/home.tsx (hunk 2 above)
Comment (Richard-Roberts) [reply client/src/components/home.tsx:20]: spanList != spanMap

File: client/src/components/sourceMapSlice.ts (hunk 3 above)
Comment (Richard-Roberts) [reply client/src/components/sourceMapSlice.ts]: initialized as undefined.

File: client/src/components/span-table/spanData.tsx (hunk 6 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: yes, removed

File: client/src/components/span-table/spanData.tsx (hunk 5 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 7 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 8 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 9 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 10 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx:49]: removed

File: client/src/components/span-table/spanData.tsx (hunk 11 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 4 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 12 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanData.tsx (hunk 13 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanData.tsx]: removed

File: client/src/components/span-table/spanHeader.tsx (hunk 14 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanHeader.tsx]: removed

File: client/src/components/span-table/spanList.tsx (hunk 15 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanList.tsx]: removed

File: client/src/components/span-table/spanList.tsx (hunk 16 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanList.tsx]: removed

File: client/src/components/span-table/spanList.tsx (hunk 17 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanList.tsx]: removed

File: client/src/components/span-table/spanList.tsx (hunk 18 above)
Comment (Richard-Roberts) [reply client/src/components/span-table/spanList.tsx]: removed

File: client/src/components/span-table/spanData.tsx [hunk 28]
```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { useSelector } from 'react-redux';
+import { useAppDispatch } from '../../lib/hooks';
+import { selectSpanDataList } from './spanDataSlice';
+import '../../styles/spanTable.scss'
+
+const spanData= (props: any) => {
+ 
+  console.log(props) 
+  console.log('props.spanData.duration', props.spanData.duration)
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx]: These logs can prolly go?

File: client/src/components/span-table/spanData.tsx [hunk 29]
```diff
@@ -0,0 +1,80 @@
...
+            {e}
+          </div>
+        </div>
+      ])
+    }
+    return result})()
+
+      const startTime = props.spanData.startTime
+
+      const time: any = new Date(startTime/1000).toString()
```
Comment (drewdunne) [inline client/src/components/span-table/spanData.tsx:62]: If you want, could wrap this around line 63 pretty easily. Also, type should just be Date or something similar.

File: client/src/components/span-table/spanDataSlice.tsx [hunk 30]
```diff
@@ -0,0 +1,66 @@
...
+//will probably need to change this from SpanNames depending on incoming data.
+type OneSpanData = SpanNames[]
+
+
+//need to update endpoint for getSpanDataAsync
+
+export const getSpanDataAsync = createAsyncThunk(
+  'spanData/getSpanData',
+  async (spanId: String) => {
+    console.log('spanId in getSpanDataAsync', spanId)
```
Comment (drewdunne) [inline client/src/components/span-table/spanDataSlice.tsx]: Is this Console left behind on purpose? Just checking.

File: client/src/components/span-table/spanDataSlice.tsx [hunk 31]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export interface SingleSpanData {
+  status: string
+  data: OneSpanData
+}
+
+type SpanNames = String
+
+//will probably need to change this from SpanNames depending on incoming data.
```
Comment (drewdunne) [inline client/src/components/span-table/spanDataSlice.tsx]: Did this end up being okay after all? Or still TBD?

File: client/src/components/span-table/spanDataSlice.tsx [hunk 32]
```diff
@@ -0,0 +1,66 @@
...
+    console.log('spanId in getSpanDataAsync', spanId)
+    const url = config.url + '/api/traces/getIndivSpanDetails/' + spanId 
+
+    //Use these logs as a first step towards troubleshooting trace fetch requests:
+    console.log("Fetching Data From: ")
+    console.log(url)
+
+    const response = await fetch(url)
+    const data = await response.json();
+    console.log(data);
```
Comment (drewdunne) [inline client/src/components/span-table/spanDataSlice.tsx]: Intended to leave this behind?

File: client/src/components/span-table/spanList.tsx [hunk 33]
```diff
@@ -0,0 +1,67 @@
...
+    dispatch(changeSpanDataView({type}))
+  }
+ 
+  const { type } = useSelector(selectSpanResultsMap)
+
+  const jsxElements = (() => {
+    const result: Array<Array<JSX.Element>> = []
+
+   
+
```
Comment (drewdunne) [inline client/src/components/span-table/spanList.tsx]: 2 extra new lines here and on 11-13.

File: client/src/components/span-table/spanResultsMap.tsx [hunk 34]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../lib/hooks';
+import SpanData from './spanData'
+import { selectSpanResultsMap, spanViewType } from './spanResultsMapSlice'
+import '../../styles/home.scss'
+
+/**
+ * Renders a specific Source Map "View" (utilizing Cytoscape)
+ * @Views : Currently Supports both Cluster View and Trace View
```
Comment (drewdunne) [inline client/src/components/span-table/spanResultsMap.tsx]: Looks like this comment isn't accurate and still here?

File: client/src/components/traceView.tsx [hunk 35]
```diff
@@ -50,29 +73,41 @@ cytoscape.use(coseBilkent);
...
+              currentTraceId = exportedtraceViewData.traceID
+              console.log('current TraceID', currentTraceId)
+              }
+
+            const traceData = {
+              processTarget : nodeData.id, 
+              traceId : currentTraceId
+            }
+
+            console.log('traceData in traceView',traceData)
```
Comment (drewdunne) [inline client/src/components/traceView.tsx]: Leftover logs?

File: client/src/styles/spanTable.scss [hunk 36]
```diff
@@ -0,0 +1,154 @@
...
+.spanHeaderPodName {
+  font-size: 1.4vh;
+  font-weight: 100;
+  padding-top: 0.7vh;
+  // padding-left: 1vw;
+}
+
+.span-pod-entry {
+  // margin-left: 1vw;
+  // margin-right: 1vw;
```
Comment (drewdunne) [inline client/src/styles/spanTable.scss]: Leftover?

**Pull request merged**